detection_rules/exception.py

```diff
@@ -64,95 +74,125 @@ class ListObject:
-    list_id: str
-    item_id: Optional[str]  # api sets field when not provided
-    meta: Optional[dict]
-    name: str
-    namespace_type: Optional[definitions.ExceptionNamespaceType]  # defaults to "single" if not provided
-    tags: Optional[List[str]]
-    type: Literal['simple']
+    field: str
+    operator: str
+    type: str
+    value: str
 
 
 @dataclass(frozen=True)
 class EndpointException(ExceptionItem, MarshmallowDataclassMixin):
     """Endpoint exception item."""
+
     _tags: List[definitions.ExceptionItemEndpointTags]
 
     @validates_schema
     def validate_endpoint(self, data: dict, **kwargs):
         """Validate the endpoint exception."""
-        for entry in data['entries']:
-            if entry['operator'] == "excluded":
+        for entry in data["entries"]:
+            if entry["operator"] == "excluded":
                 raise ValidationError("Endpoint exceptions cannot have an `excluded` operator")
 
 
 @dataclass(frozen=True)
 class DetectionException(ExceptionItem, MarshmallowDataclassMixin):
     """Detection exception item."""
+
     expire_time: Optional[str]  # fields.DateTime]  # maybe this is isoformat?
 
 
 @dataclass(frozen=True)
 class ExceptionContainer(MarshmallowDataclassMixin):
     """Exception container."""
+
     description: str
-    list_id: Optional[str]
-    meta: Optional[dict]
+    id: str
+    list_id: str
     name: str
+    entries: List[DetectionException]  # Union[DetectionException, EndpointException]]
+
+    os_types: Optional[List[str]]
     namespace_type: Optional[definitions.ExceptionNamespaceType]
+    _version: Optional[str]
     tags: Optional[List[str]]
     type: definitions.ExceptionContainerType
+    comments: Optional[List[str]]
+    created_at: Optional[str]
+    created_by: Optional[str]
+    updated_at: Optional[str]
+    updated_by: Optional[str]
+    item_id: Optional[str]  # api sets field when not provided
+    tie_breaker_id: Optional[str]
 
     def to_rule_entry(self) -> dict:
         """Returns a dict of the format required in rule.exception_list."""
         # requires KSO id to be consider valid structure
         return dict(namespace_type=self.namespace_type, type=self.type, list_id=self.list_id)
 
 
-@dataclass(frozen=True)
-class Data(MarshmallowDataclassMixin):
-    """Data stored in an exception's [exception] section of TOML."""
-    container: ExceptionContainer
-    items: List[DetectionException]  # Union[DetectionException, EndpointException]]
-
-
 @dataclass(frozen=True)
 class TOMLExceptionContents(MarshmallowDataclassMixin):
     """Data stored in an exception file."""
+
     metadata: ExceptionMeta
-    exceptions: List[Data]
+    exceptions: List[ExceptionContainer]
+
+    @classmethod
+    def from_rule_contents(
+        cls,
+        rule: TOMLRuleContents,
+        exception_data: List[dict],
+        creation_date: str = TIME_NOW,
+        updated_date: str = TIME_NOW,
```

Comment:
I would advise against generating `TIME_NOW` on the module level and using it as a default variable. Often, it's ok to do module-level constants, but the benefit should be significant (usually, we're using it as a simplistic "caching" mechanism). As a rule of thumb with module-level vars - we can't guarantee that the module will always be imported in the way we expected it to be, so we need to be mindful of the lifecycle and the value the var. Especially when the variable is as as dynamic as current datetime (even though we serialize it into a date string). 

The cleaner approach would be to make `creation_date` and `updated_date` either optional, and populate them with current time inside the function with

```python
creation_date = creation_date or datetime.now(timezone.utc).strftime("%Y/%m/%d")
```

I'm using `datetime.now(timezone.utc)` here instead of `time` to avoid all those corner cases when local non-UTC timezone can mess up the current date. These issues are so so difficult to debug when they happen.